src/accelerate/accelerator.py

```diff
@@ -1963,6 +1966,65 @@ def backward(self, loss, **kwargs):
         else:
             loss.backward(**kwargs)
 
+    def set_breakpoint(self):
+        """
+        Sets the internal flag tensor to 1 on the current process. A latter check of `Accelerator().check_breakpoint()`
+        should follow using this which will check across all processes.
+
+        Note:
+            Does not require `wait_for_everyone()`
+
+        Example:
+
+        ```python
+        >>> from accelerate import Accelerator
+
+        >>> accelerator = Accelerator()
+        >>> # Assume later in the training script
+        >>> # `should_do_breakpoint` is a custom function to monitor when to break,
+        >>> # e.g. when the loss is NaN
+        >>> if should_do_breakpoint(loss):
+        ...     accelerator.set_breakpoint()
+        >>> # Assume later in the training script
+        >>> if accelerator.check_breakpoint():
+        ...     break
+        ```
+        """
+        self.flag_tensor = torch.tensor(1, device=self.device)
+
+    def check_breakpoint(self):
+        """
+        Checks if `self.flag_tensor` has been set to 1 in any of the processes. If so, will return `True` and reset the
+        flag tensor to 0.
+
+        Note:
+            Does not require `wait_for_everyone()`
+
+        Example:
+
+        ```python
+        >>> from accelerate import Accelerator
+
+        >>> accelerator = Accelerator()
+        >>> # Assume later in the training script
+        >>> # `should_do_breakpoint` is a custom function to monitor when to break,
+        >>> # e.g. when the loss is NaN
+        >>> if should_do_breakpoint(loss):
+        ...     accelerator.set_breakpoint()
+        >>> # Assume later in the training script
+        >>> if accelerator.check_breakpoint():
+        ...     break
+        ```
+        """
+        # Now that we are outside `__init__`, we can initialize it if it is `None` on device
+        if self.flag_tensor is None:
+            self.flag_tensor = torch.tensor(0, device=self.device)
+        flag_tensor = self.reduce(self.flag_tensor)
+        if flag_tensor.item() == 1:
```

Comment:
Is it possible that the reduction operation, which is `sum` by default, returns a value greater than 1? If so, the operator should be `and`, or the check for `>=1`.

Reply:
It'd be a rare condition, but certainly something that could happen.